CONTRIBUTING.md

```diff
@@ -109,42 +115,348 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+                List<Optional<Field>> fields,
+                List<Boolean> rowIndexLocations,
+                List<String> columnNames,
+                RuntimeStats runtimeStats)
+        {
+            this.parquetReader = requireNonNull(parquetReader, "parquetReader is null");
+            this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
+            this.fields = ImmutableList.copyOf(requireNonNull(fields, "fields is null"));
+            this.rowIndexLocations = requireNonNull(rowIndexLocations, "rowIndexLocations is null");
+            this.columnNames = ImmutableList.copyOf(requireNonNull(columnNames, "columnNames is null"));
+            this.runtimeStats = requireNonNull(runtimeStats, "runtimeStats is null");
+      
+            checkArgument(
+                    types.size() == rowIndexLocations.size() && types.size() == fields.size(),
+                    "types, rowIndexLocations, and fields must correspond one-to-one-to-one");
+      
+            Streams.forEachPair(
+                    rowIndexLocations.stream(),
+                    fields.stream(),
+                    (isIndexColumn, field) -> checkArgument(
+                            !(isIndexColumn && field.isPresent()),
+                            "Field info for row index column must be empty Optional"));
+        }
+      ```
+
+* **Ordering of class members and methods**
+    * Class members are in front of methods.
+    * Group and order the class members and methods in the order of their access levels in desc order:
+        1. public
+        1. protected
+        1. package private
+           iv.	private
+    * Group and order the class members:
+        1. static final
+        1. final
+        1. normal
+    * Order the methods with the same access level in the order they’re called.
+* **Encourage proper encapsulation**
+    * Do not recommend having public class members. Use lower access levels as much as possible. Exceptions may apply for performance critical paths, in that case discuss your design first with relevant code owners.
+    * Make a function access level as low as possible
+    * If a public class is only used by one caller and the usage is very local, consider making it a nested class.
+* **Immutable and thread safe classes**
+    * Whenever possible, class fields should be final
+    * When it's not possible, ensure that accesses to the non-final fields are thread safe by whatever methods are appropriate for the circumstance (Concurrent collections, synchronized access, etc.)
+
+* **Static imports**
+  E.g. in your code, do not use something like
+
+    ```java
+    String.format("%s", rule.getClass().getSimpleName())
+    ```
+
+    * Instead, static import this method and just use “format()”
+
+        ```java
+        import static java.lang.String.format;
+        ...
+        format("%s", rule.getClass().getSimpleName())
+      
+        ```
+
+        * Similarly, do NOT use
```

Comment:
reduce one level of indentation. I guess this should be the same as line 246
the thing applies to line 262, 269, and 278